JASP-Engine/JASP/R/commonerrorcheck.R

```diff
@@ -489,12 +507,29 @@
   
   # Positive-definite?
   if (posdef && any(round(eigen(dataset)$values,10) <= 0)){
-    return(list(error=TRUE,reason="Matrix is not positive-definite"))
+    vars <- .checkForPerfectCorrelations(dataset)
+    return(list(error=TRUE,reason="Matrix is not positive-definite", vars=vars))
   }
   
   return(list(error=FALSE, reason = ""))
 }
 
+.checkForPerfectCorrelations <- function(dataset) {
+
+  vars <- list()
```

Comment:
This function looks way to much like a C like function. Loops should be avoided over vectorized operations if possible. In addition, the code only checks for correlation of 1 whereas correlations of -1 should also be picked up.
Consider:

    # some data
    p <- 3
    dat <- matrix(rnorm(20*p), 20, p)
    dat <- cbind(dat, dat[, 1])
    cr <- cor(dat)

    # a method for finding the perfectly correlated pairs
    idx <- which(lower.tri(cr, diag = FALSE), arr.ind = TRUE) # the lower triangle of the matrix
    bad <- (1 - abs(cr[idx])) <= sqrt(.Machine$double.eps)    # check if correlations are awfully close to -1 or 1
    idx[bad, ]                                                # extract perfectly correlated pairs


Whenever dealing with not strictly integer values and testing for equivalence, it's safer to test against `sqrt(.Machine$double.eps)` than against exact values. (Remember `.1 + .2 != .3`!)